### ISSUE TITLE
Number widget

### ISSUE THREAD

**Issue by stefan-korn** [GetDKAN/dkan#4068]
## User Story
Would be nice to have a number widget available for specific schema properties.

Proposal:
```
"ui:options" {
  "widget": "number",
  "step": "0.01",
  "min": "10",
  "max": "20"
}
```
step, min and max are optional.

**Comment by dafeder** [comment]
How would this be different from a text widget? Would we possibly have issues here with the ui options conflicting with the schema validation? Open to this just want to understand the use-case and implications better.

**Comment by stefan-korn** [comment]
@dafeder : it would provide a [number widget](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21Element%21Number.php/class/Number/10), that has some numerical validation and the "stepper". I consider this an UI improvement if you want to collect numerical input.
Regarding schema validation I am not sure: The value being saved in the json data should not differ from collecting via text widget, except that only numbers are allowed. So there should be no problem validating.
If there is also schema validation for schema.ui.json too, there might be problems. But I tested this change and did not have problems so far.

**Comment by dafeder** [comment]
There really should be a spec for the UI schema both for validation and documentation purposes, it's a longtime todo. But yeah for now I see the value of this.

**Comment by dafeder** [comment]
@stefan-korn I did another review of your PR on this, interested in your response and whether you have some capacity to work more on this.

**Comment by dafeder** [comment]
Resolved in #4215
